tools/scenario-player/scenario_player/tasks.py

```diff
@@ -0,0 +1,336 @@
+                ]
+            ],
+        )
+
+
+class CloseChannelTask(ChannelActionTask):
+    _name = 'close_channel'
+
+    @property
+    def _request_params(self):
+        return dict(state='closed')
+
+
+class DepositTask(ChannelActionTask):
+    _name = 'deposit'
+
+    @property
+    def _request_params(self):
+        return dict(total_deposit=self._config['total_deposit'])
+
+
+class TransferTask(ChannelActionTask):
+    _name = 'transfer'
+    _url_template = '{protocol}://{target_host}/api/1/payments/{token_address}/{partner_address}'
+    _method = 'post'
+
+    @property
+    def _request_params(self):
+        return dict(amount=self._config['amount'])
+
+
+class WaitTask(Task):
+    def _run(self, *args, **kwargs):
+        gevent.sleep(self._config)
+
+
+class AssertTask(ChannelActionTask):
+    _name = 'assert'
+    _method = 'get'
+
+    def _process_response(self, response: dict):
+        for field in ['balance', 'total_deposit', 'state']:
+            if field not in self._config:
+                continue
+            if field not in response:
+                raise ScenarioAssertionError(f'Field "{field}" is missing in channel: {response}')
+            if response[field] != self._config[field]:
+                raise ScenarioAssertionError(
+                    f'Value mismatch for "{field}". '
+                    f'Should: "{self._config[field]}" '
+                    f'Is: "{response[field]}" '
+                    f'Channel: {response}',
+                )
+
+
+class ProcessTask(Task):
+    _command = ''
+
+    def _run(self, *args, **kwargs):
+        command = self._runner.node_commands.get(self._command)
+        if not command:
+            raise ScenarioError(
+                'Invalid scenario definition. '
+                f'The {self._command}_node task requires '
+                f'nodes.commands.{self._command} to be set.',
+            )
+        command = command.format(self._config)
+        log.debug('Command', type_=self._command, command=command)
+        greenlet = gevent.spawn(subprocess.run, shlex.split(command), check=True)
+        self._handle_process(greenlet)
+
+    def _handle_process(self, greenlet):
+        greenlet.join()
+        greenlet.get()
+
+
+class StartNodeTask(ProcessTask):
+    _command = 'start'
+
+    def _handle_process(self, greenlet):
+        # FIXME: Wait for port to become available and then stop blocking on the greenlet
+        super()._handle_process(greenlet)
+
+
+class StopNodeTask(ProcessTask):
+    _command = 'stop'
+
+
+class KillNodeTask(ProcessTask):
+    _command = 'kill'
+
+
+NAME_TO_TASK = {
+    'serial': SerialTask,
+    'parallel': ParallelTask,
+    'open_channel': OpenChannelTask,
+    'close_channel': CloseChannelTask,
+    'deposit': DepositTask,
+    'transfer': TransferTask,
+    'wait': WaitTask,
+    'assert': AssertTask,
+    'stop_node': StopNodeTask,
+    'start_node': StartNodeTask,
+    'kill_node': KillNodeTask,
```

Comment:
Since we're going to be writing more tasks... and most probably moving `tasks.py` into a folder where sub-modules exist with different types of tasks, we could implement auto-discovery of tasks and programatically build this dictionary. Tasks seem to be already specifying their names (example: [here](https://github.com/raiden-network/raiden/pull/2524/files#diff-ab74835b3e876a03a4b44d4e8bb56abdR259)) which could be used for the key part of the dict.